src/coins.cpp

```diff
@@ -562,3 +552,37 @@ CoinModifier::~CoinModifier()
     cache->cs_utxo.unlock();
     LeaveCritical();
 }
+
+void AddCoins(CCoinsViewCache &cache, const CTransaction &tx, int nHeight)
+{
+    bool fCoinbase = tx.IsCoinBase();
```

Comment:
not sure what these functions from coin.cpp has to do with the main.cpp refactor?

Reply:
AddCoins was just moved to the bottom of the file (although i think that was unintentional but i dont see how it matters). updatecoins was moved from main.cpp to coins.cpp because it seemed out of place that this one coins function was just left in main after i moved the other code to validation instead of with the rest of the coin related functions.

Reply:
ah, ok i didn't see where updatecoins was coming from.